api/v3/Job/Eway.php

```diff
@@ -90,23 +95,17 @@ function _civicrm_api3_job_eway_process_contribution($instance) {
     // Process the contribution as either Completed or Failed.
     $apiResult[] = "Successfully processed payment for " . $instance['type'] . " recurring contribution ID: " . $instance['contribution_recur']->id;
     $apiResult[] = "Marking contribution as complete";
-    $instance['contribution']->trxn_id = $result['values'][$managed_customer_id]['trxn_id'];
-    if (empty($instance['contribution']->id)) {
-      repeat_contribution($instance['contribution'], 'Completed', $amount_in_cents);
-    }
-    else {
-      complete_contribution($instance['contribution']);
-    }
+    $pendingContribution['values'][$pendingContribution['id']]['trxn_id'] = $result['values'][$managed_customer_id]['trxn_id'];
+    complete_contribution($pendingContribution['values'][$pendingContribution['id']]);
     $instance['contribution_recur']->failure_count = 0;
   }
   catch (CiviCRM_API3_Exception $e) {
     $apiResult[] = "ERROR: failed to process payment for " . $instance['type'] . " recurring contribution ID: " . $instance['contribution_recur']->id;
     $apiResult[] = 'eWAY managed customer: ' . $instance['contribution_recur']->processor_id;
     $apiResult[] = 'eWAY response: ' . $result['faultstring'];
```

Comment:
I changed this line too. `$result['faultstring']` becomes `$e->getMessage()`

Reply:
`$result` is never populated in the case that an error is thrown.

Reply:
thanks @andrew-cormick-dockery updated it now